src/CarbonAware.WebApi/test/integrationTests/CarbonAwareControllerTests.cs

```diff
@@ -184,39 +186,143 @@ public async Task EmissionsForecastsBatch_MissingRequiredParams_ReturnsBadReques
         {
-            new EmissionsForecastBatchDTO
+            Assert.That(result, Is.Not.Null);
+            Assert.That(result.StatusCode, Is.EqualTo(HttpStatusCode.OK));
+            Assert.That(result!.Content, Is.Not.Null);
+            using (var data = await result!.Content.ReadAsStreamAsync())
             {
-                Location = "eastus",
-                DataStartAt = new DateTimeOffset(2021,9,1,8,30,0, TimeSpan.Zero),
-                DataEndAt = new DateTimeOffset(2021,9,2,8,30,0, TimeSpan.Zero),
-                RequestedAt = new DateTimeOffset(2021,9,1,8,30,0, TimeSpan.Zero)
+                Assert.That(data, Is.Not.Null);
+                var value = await JsonSerializer.DeserializeAsync<CarbonIntensityDTO>(data);
+                Assert.That(value, Is.Not.Null);
+                Assert.That(value!.CarbonIntensity, Is.Not.EqualTo(0));
+                Assert.That(value!.StartTime, Is.EqualTo(start));
+                Assert.That(value!.EndTime, Is.EqualTo(end));
             }
-        };
+        }
+    }
+
+    [TestCase("location", "", TestName = "EmissionsActual empty location query string expects BadRequest")]
+    [TestCase("non-location-param", "", TestName = "EmissionsActual location param not present expects BadRequest")]
+    public async Task EmissionsActual_EmptyLocationQueryString_ReturnsBadRequest(string queryString, string value)
+    {
+        var queryStrings = new Dictionary<string, string>();
+        queryStrings[queryString] = value;
 
-        var result = await PostJSONBodyToURI(inputData, batchForecastURI);
+        var endpointURI = ConstructUriWithQueryString(actualHistoricalURI, queryStrings);
+        var result = await _client.GetAsync(endpointURI);
+
+        Assert.That(result, Is.Not.Null);
+        Assert.That(result.StatusCode, Is.EqualTo(HttpStatusCode.BadRequest));
+    }
+
+    [TestCase(false, false, false, TestName = "EmissionsBatchActual Not use location, Not use startTime, Not use endTime expects BadRequest")]
+    [TestCase(true, false, false, TestName = "EmissionsBatchActual Use location, Not use startTime, Not use endTime expects BadRequest")]
+    [TestCase(true, true, false, TestName = "EmissionsBatchActual Use location, Use startTime, Not use endTime expects BadRequest")]
+    public async Task EmissionsBatchActual_MissingRequiredParams_ReturnsBadRequest(bool useLocation, bool useStart, bool useEnd)
+    {
+        if (useLocation && useStart && useEnd)
+        {
+            Assert.Fail("Invalid test");
+        }
+
+        var intensityBatch = new CarbonIntensityBatchDTO();
+        intensityBatch.Location = useLocation ? "eastus" : null;
+        intensityBatch.StartTime = useStart ? DateTimeOffset.Parse("2022-03-01T15:30:00Z") : null;
+        intensityBatch.EndTime = useEnd ? DateTimeOffset.Parse("2022-03-01T18:30:00Z") : null;
        
+        var intesityData = new List<CarbonIntensityBatchDTO>() { intensityBatch };
```

Comment:
Similar to other methods, we should create a JSON payload that does not know about internal models